pfhedge/_utils/bisect.py

```diff
@@ -110,4 +110,11 @@ def find_implied_volatility(
     def fn(volatility: Tensor) -> Tensor:
         return pricer(volatility=volatility, **params)
 
-    return bisect(fn, price, lower, upper, precision=precision, max_iter=max_iter)
+    return bisect(
+        fn,
+        price,
+        torch.as_tensor(lower, device=price.device),
+        torch.as_tensor(upper, device=price.device),
```

Comment:
dtypeも普通は揃えたい気がするので、その時はこっちの方がシンプル

```suggestion
        torch.as_tensor(lower).to(price),
        torch.as_tensor(upper).to(price),
```